src/Proto.Cluster/Cluster.cs

```diff
@@ -67,13 +67,19 @@ private static (string host, int port) ParseAddress(string address)
             return (host, port);
         }
 
-        public static Task<PID> GetAsync(string name, string kind) => GetAsync(name, kind, CancellationToken.None);
+        public static Task<(PID, ResponseStatusCode)> GetAsync(string name, string kind) => GetAsync(name, kind, CancellationToken.None);
 
-        public static async Task<PID> GetAsync(string name, string kind, CancellationToken ct)
+        public static async Task<(PID, ResponseStatusCode)> GetAsync(string name, string kind, CancellationToken ct)
         {
             var req = new PidCacheRequest(name, kind);
-            var res = await PidCache.Pid.RequestAsync<ActorPidResponse>(req, ct);
-            return res.Pid;
+            var res = await PidCache.Pid.RequestAsync<PidCacheResponse>(req, ct);

```

Comment:
Is it possible that the ProcessNameExists bubbles up here?
And if so, should we not return the existing PID in that case?

To me it looks like if two different threads/nodes at the same time tries to `GetAsync("SomeActor","SomeKind")` one of them might end up getting ProcessNameExists back, right?

Reply:
Hi, you are right! Actually in case of ProcessNameExists, req.Pid will return null since Activator will only assign status code. 
https://github.com/PotterDai/protoactor-dotnet/blob/1063da1bf8c08d4bce461f86c9205fba14440bbd/src/Proto.Remote/Activator.cs#L45-L49

Reply:
Maybe it is enough to extend ActorPidResponse with the status code and do what you say?
This way, GetAsync would behave as expected no matter if it is OK or ProcessNameExists

Reply:
OK, so I think it should be good to go! = ) Any exception now will return with a null PID and a not-OK StatusCode, user can decide what to do accordingly.